qprofiler2/src/org/qcmg/qprofiler2/util/XmlUtils.java

```diff
@@ -236,9 +239,25 @@ public static Element outputBins( Element parent, String name, Map<Integer, Atom
     		}  	    		
     	}
     	
-    	return cateEle;
-    	
+    	return cateEle;   	
     }
+	
+	public static <T> void outputTallyGroupWithSize(Element parent, String name, Map<T, AtomicLong> tallys, int sizeLimits) {
+		String comment = null; 
+		boolean hasPercent =  true;
+		String tallyCount = null; 
+		if( tallys.size() > sizeLimits) { 			 
```

Comment:
what if `tallys` is null? Should always check inputs of static methods

Reply:
done